### PR TITLE
Fix Rigetti demonstrations

### PR DESCRIPTION
**Pull request by josh146**
With the latest update to the QCS, QPU lattices are no longer valid QVM devices. In addition, the Aspen-7 device has been removed, and the Aspen-4 device will be removed May 31, leaving only the new 32Q Aspen-8 device.

**Changes:**

* Upgrades Sphinx-Gallery to v0.7, to address backreference bugs.

* Pins PL-Forest to GitHub master, which involves bugfixes related to the above Rigetti changes.

* Modifies the parallel VQE and ensemble multi-QPU demo.

### CODE REVIEW

**Comment by josh146** [comment]
* Ensemble multi-QPU demo passing ✔️ 

* Parallel VQE failing, since instantiating `Aspen-8` on the QVM consumes too much memory, causing a crash ❌  ([traceback here](https://app.circleci.com/pipelines/github/XanaduAI/qml/759/workflows/d0441940-0a53-40f9-a456-c36698613904/jobs/785/tests))

File: demonstrations/tutorial_vqe_parallel.py [hunk 1]
```diff
@@ -110,23 +110,30 @@
...
 #    :align: center
 #
 # To do this, start by instantiating a device for each term:
 
-devs_4 = [qml.device("forest.qvm", device="Aspen-4-4Q-F") for _ in range(8)]
-devs_7 = [qml.device("forest.qvm", device="Aspen-7-4Q-D") for _ in range(7)]
-devs = devs_4 + devs_7
+devs_aspen_8 = [qml.device("forest.qvm", device="Aspen-8") for _ in range(8)]
+devs_4q_qvm = [qml.device("forest.qvm", device="4q-qvm") for _ in range(7)]
+devs = devs_aspen_8 + devs_4q_qvm
```
Comment (josh146) [inline demonstrations/tutorial_vqe_parallel.py:125]:
```suggestion
dev1 = [qml.device("forest.qvm", device="4q-qvm") for _ in range(8)]
dev2 = [qml.device("forest.qvm", device="4q-qvm") for _ in range(7)]
devs = dev1 + dev2
```

File: demonstrations/tutorial_vqe_parallel.py [hunk 2]
```diff
@@ -110,23 +110,30 @@
...
 # .. note::
-#     You can swap out ``forest.qvm`` for ``forest.qpu`` if hardware access is available.
+#
+#     For the purposes of this demonstration, we are simulating the QPUs using the
+#     ``forest.qvm`` simulator. To run this demonstration on hardware, simply
+#     swap ``forest.qvm`` for ``forest.qpu``, and make sure to use a valid QPU
+#     identifier.
+#
+#     Please refer to the `Rigetti website <https://rigetti.com/>`__ for an up-to-date
+#     list on available QPUs.
```
Comment (josh146) [inline demonstrations/tutorial_vqe_parallel.py:136]: @trbromley: Just checking for your thoughts on this notebox. I'm not sure it does quite a good job explaining why `devs` is essentially just a list of `4q-qvm` :(

Should we make the second group `5q-qvm` for demonstration?

Feel free to make any changes as you wish here.

**Comment by josh146** [comment]
<h3>Website build</h3>
<strong>Commit:</strong> fc2b74d8a7ababf39b15eebaa46ffbc0f45f4507

<strong>Circle build number:</strong> 789

<strong>Built <code>demos</code> directory:</strong> <a href=https://789-214003948-gh.circle-artifacts.com/0/tmp/qml_demos.zip>https://789-214003948-gh.circle-artifacts.com/0/tmp/qml_demos.zip</a>

<strong>Backreferences zip:</strong> <a href=https://789-214003948-gh.circle-artifacts.com/0/tmp/qml_backreferences.zip>https://789-214003948-gh.circle-artifacts.com/0/tmp/qml_backreferences.zip</a>

<strong>Website zip:</strong> <a href=https://789-214003948-gh.circle-artifacts.com/0/tmp/qml_html.zip>https://789-214003948-gh.circle-artifacts.com/0/tmp/qml_html.zip</a>

<strong>Website url:</strong> <a href=https://789-214003948-gh.circle-artifacts.com/0/_build/html/index.html>https://789-214003948-gh.circle-artifacts.com/0/_build/html/index.html</a>

<em>Please double check the rendered website build to make sure everything is correct.</em>

File: demonstrations/tutorial_vqe_parallel.py (hunk 2 above)
Comment (trbromley) [reply demonstrations/tutorial_vqe_parallel.py]: I had a go at updating this tutorial and adding a sentence to the other one + updating the images, would be good to see what you think!

**Pull request merged**